database/altsig.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package database
+
+import (
+	"errors"
+	"fmt"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+const (
+	// maxAltSigs is the maximum number of signature history to allow into
+	// the database for a ticket. It is necessary to limit records to avoid
+	// remote attackers filling the database with arbitrary data.
+	maxAltSigs = 1
+)
+
+// ErrMaxAltSigs is returned when attempting to update a ticket's alternate
+// signature but maxAltsSigs or more history is already in the database.
+var ErrMaxAltSigs = errors.New("max alt sigs")
+
+// The keys used to store the altsig in the database.
+var (
+	altSigAddrK = []byte("altsig")
+
+	altSigHistoryK     = []byte("altsighistory")
+	altSigHistoryAddrK = []byte("altsighistoryaddr")
+	altSigHistoryReqK  = []byte("altsighistoryreq")
+	altSigHistorySigK  = []byte("altsighistorysig")
+)
+
+// AltSigHistory holds the information needed to prove that a client changed
+// their alternate signature address.
+type AltSigHistory struct {
+	// Addr is the address used to validate Sig.
+	Addr string
+	// Req is the original request to set an alternate signature.
+	Req []byte
+	// Sig is the request's signature signed by the private key that
+	// corresponds to the address.
+	Sig []byte
+}
+
+// InsertAltSig will insert the provided ticket into the database. Returns an
+// error if either the ticket hash or fee address already exist.
+//
+// history must be included.
+func (vdb *VspDatabase) InsertAltSig(txid, altSigAddr string, history *AltSigHistory) error {
```

Comment:
```suggestion
func (vdb *VspDatabase) InsertAltSig(ticketHash, altSigAddr string, history *AltSigHistory) error {
```

I know its more verbose, but for the sake of maintaining consistency with the rest of the codebase, would like to stick with the more descriptive `ticketHash` here please.

Same for all other DB funcs.

Reply:
Ok.